crates/llm-base/src/inference_session.rs

```diff
@@ -26,6 +26,24 @@ fn scratch_buffers() -> ScratchBuffers {
     ]
 }
 
+fn kv_memory(
```

Comment:
Helper functions at the bottom; the first thing a user should see looking at the file is the public interface.